lib/_tls_wrap.js

```diff
@@ -571,6 +571,27 @@ tls_wrap.TLSWrap.prototype.close = function close(cb) {
   return this._parent.close(done);
 };
 
+TLSSocket.prototype.destroySoon = function() {
+  if (this.writable)
+    this.end();
+
```

Comment:
I think `this.end()` is enough here. The code below is possible redundant.

Reply:
Also should we have a timeout waiting for `'close'` here? @lpinca

Reply:
Normally, the kernel will expire sockets stuck in `FIN_WAIT_2` state as it has always been a problem and there was an old DoS attack that starved out the system of sockets by deliberately not closing them